opal/mca/accelerator/rocm/accelerator_rocm_module.c

```diff
@@ -158,7 +166,8 @@ static int mca_accelerator_rocm_create_stream(int dev_id, opal_accelerator_strea
 
 static void mca_accelerator_rocm_stream_destruct(opal_accelerator_rocm_stream_t *stream)
 {
-    if (NULL != stream->base.stream) {
+    if (MCA_ACCELERATOR_STREAM_DEFAULT != (opal_accelerator_stream_t *)stream &&
+	NULL != stream->base.stream) {
```

Comment:
yes, it will need the same change

Reply:
Could you please do it in this change in case we forget?

Reply:
I added the changes. It might not have been super necessary in the case of cuda because the cuda component didn't check for stream->stream being NULL, but I added it anyway just to be safe. I also added the GET_STREAM logic to cuda component as well.

Reply:
@hppritcha I was looking into the ze component and whether I can apply the same changes to this component, but I honestly do not know enough about level zero API, and e.g. how the concept of the default stream would be handled there. I am however more than happy to work with you to resolve this, please feel free to ping me for that.

Reply:
i will take a look at ways to do this in ZE, but will have to wait till next week.